src/platform/plugins/shared/controls/public/controls/esql_control/get_esql_control_factory.test.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React from 'react';
+import { BehaviorSubject } from 'rxjs';
+import { StateComparators } from '@kbn/presentation-publishing';
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import { getMockedControlGroupApi } from '../mocks/control_mocks';
+import type { ControlApiRegistration } from '../types';
+import { getESQLControlFactory } from './get_esql_control_factory';
+import type { ESQLControlApi, ESQLControlState } from './types';
+
+describe('ESQLControlApi', () => {
+  const uuid = 'myESQLControl';
+
+  const dashboardApi = {};
+  const controlGroupApi = getMockedControlGroupApi(dashboardApi);
+
+  const factory = getESQLControlFactory();
+  function buildApiMock(
+    api: ControlApiRegistration<ESQLControlApi>,
+    nextComparators: StateComparators<ESQLControlState>
+  ) {
+    return {
+      ...api,
+      uuid,
+      parentApi: controlGroupApi,
+      unsavedChanges: new BehaviorSubject<Partial<ESQLControlState> | undefined>(undefined),
+      resetUnsavedChanges: () => {
+        return true;
+      },
+      type: factory.type,
+    };
+  }
+
+  test('Should add a new variable to the ES|QL service derived from state', async () => {
+    const initialState = {
+      selectedOptions: ['option1'],
+      availableOptions: ['option1', 'option2'],
+      variableName: 'variable1',
+      variableType: 'values',
+      esqlQuery: 'FROM foo | WHERE column = ?variable1',
+      controlType: 'STATIC_VALUES',
+    } as ESQLControlState;
+    const { api } = await factory.buildControl(initialState, buildApiMock, uuid, controlGroupApi);
+    expect(api.esqlVariable$.value).toStrictEqual({
+      key: 'variable1',
+      type: 'values',
+      value: 'option1',
+    });
+  });
+
+  test('Should get the serialized state correctly', async () => {
```

Comment:
How about just "Should serialize state"

Reply:
Done https://github.com/elastic/kibana/pull/202875/commits/13531c5e8138ff01611039d7e9bd8810af77a399